app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,64 @@
+class CommentsController < ApplicationController
+
+  def new
+    @question = Question.find(params[:question_id])
+    @comment = @question.comments.new
+  end
+
+  def create
+    params[:question_id] ? question_comment : answer_comment
```

Comment:
I'd maybe have answer_comment and question_comment return the new_comment and set @new_comment here. Calling a method to set and ivar can be difficult to follow.